src/network.c

```diff
@@ -685,7 +685,7 @@ size_t readn_ssl(SSL *ssl, void *buf, size_t amount) {
  * @param fd The file descriptor to read from
  * @param buf The buffer to read into
  * @param amount The the amount of data that can be read into buf
- * @return The number of bytes read
+ * @return The number of bytes read, or a negative error code.
  */
 size_t readn_raw(int fd, void *buf, size_t amount) {
```

Comment:
They wanted to add the ability to timeout on `readn`, and so they needed `select()`.  Depending on who you believe on the Internet, the `BIO` objects underlying the `SSL_read` function should never block, and thus we have no need for `select()` prior to `SSL_read`. This may or may not be true, though. IIRC, we already found one instance of "I thought the documentation said X about nonblocking BIOs, but it could be interpreted to mean Y, I suppose". I wrote this code prior to our discovery that the documentation was so perniciously unclear.